kubespawner/spawner.py

```diff
@@ -2215,10 +2219,15 @@ def load_state(self, state):
         but if the `pod_template` has changed in between restarts, it will no longer
         be the case. This allows us to continue serving from the old pods with
         the old names.
+
+        For a similar reason, we also save the namespace.
         """
         if 'pod_name' in state:
             self.pod_name = state['pod_name']
 
+        if 'namespace' in state:
```

Comment:
Actually, a lot of other variables are already set in `__init__` with an old namespace value:
https://github.com/jupyterhub/kubespawner/blob/acfc4d5efc364b8eff07efbe83da3904c3a80435/kubespawner/spawner.py#L180-L188
Restoring only `namespace` could cause issues with them

Reply:
Ah, you're right. I think in many cases, this doesn't manifest as an actual bug because some properties are only used to create the pod, and since we're talking about serialization here, the pod has already been created.

The values are still wrong after deserialization and might cause problems.

I guess we could either also save the dependent attributes, or recalculate them after restore.

I think saving those attribute could be more robust because e.g. the `dns_name` could have been changed since the pod has been created, but the existing pod will still have the old value no matter what.

What do you think?

Reply:
Created #677 677